hivemq4/base-image/entrypoints.d/30_configure_permissions.sh

```diff
@@ -24,14 +24,14 @@ if [[ "$(id -u)" = "0" ]]; then
     chown "${uid}":"${gid}" /opt
     chown "${uid}":"${gid}" /opt/hivemq
     chown "${uid}":"${gid}" /opt/hivemq-*
-    chown "${uid}":"${gid}" /opt/hivemq/audit
-    chown "${uid}":"${gid}" /opt/hivemq/log
-    chown "${uid}":"${gid}" /opt/hivemq/conf
-    chown "${uid}":"${gid}" /opt/hivemq/conf/config.xml
-    chown "${uid}":"${gid}" /opt/hivemq/license
-    chown "${uid}":"${gid}" /opt/hivemq/backup
-    chown "${uid}":"${gid}" /opt/hivemq/tools
-    chown "${uid}":"${gid}" /opt/hivemq/extensions
+    chown -R "${uid}":"${gid}" /opt/hivemq/audit
+    chown -R "${uid}":"${gid}" /opt/hivemq/log
+    chown -R "${uid}":"${gid}" /opt/hivemq/conf
+    chown -R "${uid}":"${gid}" /opt/hivemq/conf/config.xml
+    chown -R "${uid}":"${gid}" /opt/hivemq/license
+    chown -R "${uid}":"${gid}" /opt/hivemq/backup
+    chown -R "${uid}":"${gid}" /opt/hivemq/tools
+    chown -R "${uid}":"${gid}" /opt/hivemq/extensions
```

Comment:
Is there a specific reason that we chown individual folders instead of doing a `chown -R "${uid}":"${gid}" /opt/hivemq` ?

Reply:
Tried it out:
```
 chown "${uid}":"${gid}" /opt/hivemq-*
 chown -R "${uid}":"${gid}" /opt/hivemq
 # Recursive for bin, no volume here
```

Outcome:
```
2021-09-14 10:12:34,289 WARN  - Not able to watch extension folder for changes, reason: /opt/hivemq/extensions. Disabling runtime reloading of extensions.
2021-09-14 10:12:34,295 ERROR - Uncaught Error:
java.lang.IllegalArgumentException: /opt/hivemq/extensions is not readable
        at com.google.common.base.Preconditions.checkArgument(Unknown Source)
        at dw.i.a(Unknown Source)
        at cO.h.a(Unknown Source)
        at com.hivemq.a.b(Unknown Source)
        at com.hivemq.a.a(Unknown Source)
        at com.hivemq.HiveMQServer.b(Unknown Source)
        at com.hivemq.HiveMQServer.d(Unknown Source)
        at com.hivemq.HiveMQServer.main(Unknown Source)

```